occurrence-download/src/main/resources/templates/species-list-download/execute-species-list-query.ftl

```diff
@@ -0,0 +1,44 @@
+<#--
+  This is a freemarker template which will generate an HQL script which is run at download time.
+  When run in Hive as a parameterized query, this will create a set of tables ...
+  TODO: document when we actually know something accurate to write here...
+-->
+<#-- Required syntax to escape Hive parameters. Outputs "USE ${hiveDB};" -->
+USE ${r"${hiveDB}"};
+
+-- setup for our custom, combinable deflated compression
+SET hive.exec.compress.output=true;
+SET io.seqfile.compression.type=BLOCK;
+SET mapred.output.compression.codec=org.gbif.hadoop.compress.d2.D2Codec;
+SET io.compression.codecs=org.gbif.hadoop.compress.d2.D2Codec;
+
+-- in case this job is relaunched
+DROP TABLE IF EXISTS ${r"${speciesListTable}"};
+DROP TABLE IF EXISTS ${r"${speciesListTable}"}_tmp;
+DROP TABLE IF EXISTS ${r"${speciesListTable}"}_citation;
+
+-- pre-create verbatim table so it can be used in the multi-insert
+CREATE TABLE ${r"${speciesListTable}"}_tmp ROW FORMAT DELIMITED FIELDS TERMINATED BY '\t'
+AS SELECT
+taxonkey , scientificname, taxonrank, taxonomicstatus, kingdom, kingdomkey, phylum, phylumkey,class,classkey, order_, orderkey, family,familykey, genus,genuskey, subgenus, subgenuskey, species, specieskey , datasetkey
+FROM occurrence_hdfs
+WHERE ${r"${whereClause}"};
+
+CREATE TABLE ${r"${speciesListTable}"} ROW FORMAT DELIMITED FIELDS TERMINATED BY '\t'
+TBLPROPERTIES ("serialization.null.format"="")
+AS SELECT
+taxonkey , scientificname, count(taxonkey) as no_of_occurrences, taxonrank, taxonomicstatus, kingdom, kingdomkey, phylum, phylumkey,class,classkey, order_, orderkey, family,familykey, genus,genuskey, subgenus, subgenuskey, species, specieskey
+FROM ${r"${speciesListTable}"}_tmp
+GROUP BY 
+taxonkey, scientificname, taxonrank, taxonomicstatus, kingdom, kingdomkey,phylum, phylumkey, class,classkey,order_, orderkey, family, familykey, genus, genuskey , subgenus, subgenuskey, species, specieskey;
```

Comment:
Does taxonRank get populated from a nub lookup call always? If not, this might be filled with nonsense... I don't recall how we populate that field